src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -223,6 +223,10 @@ private static bool IsTupleExpression(BoundKind kind)
             {
                 var field = fields[i];
 
+                // Use default field rather than implicitly named fields since
+                // fields from inferred names are not usable in C# 7.0.
+                field = field.CorrespondingTupleField ?? field;
```

Comment:
use-site diagnostic here is likely for cases with broken/malformed ValueTuple - like when the corresponding field is missing or inaccessible for other reasons, we should produce an error not a crash or something like that.